### ISSUE TITLE
Environment markers don't work on pip 8.1.1

### ISSUE THREAD

**Issue by samkhal** [jazzband/pip-tools#1027]
When using pip 8.1.1, environment markers don't work.
Solution: upgrade to pip 19.3.1.

Given the plan to stop supporting old python versions in https://github.com/jazzband/pip-tools/issues/926, this probably doesn't need to be fixed.

#### Environment Versions

1. OS Type: Ubuntu 16.04
1. Python version: 2.7.12
1. pip version:  8.1.1
1. pip-tools version: 4.3.0

#### Steps to replicate

1. `echo "numpy; python_version<'3.0'" > requirements.in`
2. `pip-compile requirements.in`

#### Expected result
`requirements.txt` should contain `numpy==1.16.6 ; python_version < '3.0'`

#### Actual result
```
Traceback (most recent call last):
  File "/home/sam/.virtualenvs/py2_sandbox/bin/pip-compile", line 8, in <module>
    sys.exit(cli())
  File "/home/sam/.virtualenvs/py2_sandbox/local/lib/python2.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/home/sam/.virtualenvs/py2_sandbox/local/lib/python2.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/home/sam/.virtualenvs/py2_sandbox/local/lib/python2.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/sam/.virtualenvs/py2_sandbox/local/lib/python2.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/home/sam/.virtualenvs/py2_sandbox/local/lib/python2.7/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/sam/.virtualenvs/py2_sandbox/local/lib/python2.7/site-packages/piptools/scripts/compile.py", line 338, in cli
    req for req in constraints if req.markers is None or req.markers.evaluate()
AttributeError: 'unicode' object has no attribute 'evaluate'
```

**Comment by atugushev** [comment]
Hello @samkhal,

Thanks for the issue! This is the pip's bug which was fixed in `pip==8.1.2`.

**Comment by atugushev** [comment]
Closed based on the above answer.
